File: js-archive/number-of-arithmetic-triplets.js
const arithmeticTriplets = function (nums, diff) {
  let ans = 0
  for (let i = 0; i < nums.length - 2; i++) {
    for (let j = i + 1; j < nums.length - 1; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        if (nums[j] - nums[i] === diff &&
          nums[k] - nums[j] === diff
        ) {
          ans++
        }
      }
    }
  }

  return ans
}

console.log(arithmeticTriplets([0, 1, 4, 6, 7, 10], diff = 3))

File: number-of-arithmetic-triplets.go
func arithmeticTriplets(nums []int, diff int) int {
    ans:=0
    for i:=0; i < len(nums); i++ {
        for j:=i + 1; j < len(nums); j++ {
            for k:=j + 1; k < len(nums); k++ {
                if nums[i] + diff == nums[j] && nums[j] + diff == nums[k] {ans++}
            }
        }
    }

    return ans
}